geode-core/src/main/java/org/apache/geode/distributed/internal/membership/adapter/GMSMembershipManager.java

```diff
@@ -840,11 +782,14 @@ private void removeWithViewLock(InternalDistributedMember dm, boolean crashed, S
    * @param member the member
    */
   protected void handleOrDeferSurpriseConnect(InternalDistributedMember member) {
-    synchronized (startupLock) {
+    startupLock.readLock().lock();
```

Comment:
It looks like this PR is also changing the way this class does synchronization. I'm not sure we want to combine these synchronization related changes with refactoring the dependencies of the membership code.

Reply:
I've removed that change.  It needs to be done sometime though - synchronizing on this object for every message dispatched has got to be hurting performance.